iron-multi-selectable.html

```diff
@@ -81,11 +81,18 @@
     },
 
     _selectMulti: function(values) {
-      this._selection.clear();
       if (values) {
-        for (var i = 0; i < values.length; i++) {
-          this._selection.setItemSelected(this._valueToItem(values[i]), true);
+        var selectedItems = this._valuesToItems(values);
+        // clear all but the current selected items
+        this._selection.clear(selectedItems);
+        // select only those not selected yet
+        for (var i = 0; i < selectedItems.length; i++) {
+          if (!this._selection.isSelected(selectedItems[i])) {
```

Comment:
I think this conditional should be moved into [`IronSelection.setItemSelected`](https://github.com/PolymerElements/iron-selector/blob/a2a54b651faf48503d7196203fc6df7ae03bf81e/iron-selection.html#L73) (probably as `this._selection.indexOf(...) < 0`). For some reason, `IronSelection.setItemSelected`'s current implementation is inconsistent / lame: it's ok with trying to deselect an item that's not selected but it breaks when trying to select something that's already selected.